wire-ios-sync-engine/Source/UserSession/ZMUserSession+RecurringAction.swift

```diff
@@ -0,0 +1,56 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+extension ZMUserSession {
+
+    func refreshUsersMissingMetadata() -> RecurringAction {
+        /// Performing actions every 3 hours.
+        let interval: TimeInterval = 3 * 60 * 60
+
+        return RecurringAction(id: UUID().uuidString, interval: interval) {
```

Comment:
Every time you create a new action it will have a different ID. This ID is persisted in the user defaults which will persist data between launches. This means you'll sometimes have no last check date because you're querying with a new id.

I suggest to hardcode an id name, like `refreshUserMetadata`

Reply:
done